src/Servers/Kestrel/Core/src/Internal/ConfigurationReader.cs

```diff
@@ -73,14 +87,25 @@ private IEnumerable<EndpointConfig> ReadEndpoints()
             foreach (var endpointConfig in endpointsConfig)
             {
                 // "EndpointName": {
-                //    "Url": "https://*:5463",
-                //    "Protocols": "Http1AndHttp2",
-                //    "SslProtocols": [ "Tls11", "Tls12", "Tls13"],
-                //    "Certificate": {
-                //        "Path": "testCert.pfx",
-                //        "Password": "testPassword"
-                //    },
-                //    "ClientCertificateMode" : "NoCertificate"
+                //     "Url": "https://*:5463",
+                //     "Protocols": "Http1AndHttp2",
+                //     "SslProtocols": [ "Tls11", "Tls12", "Tls13"],
+                //     "Certificate": {
+                //         "Path": "testCert.pfx",
+                //         "Password": "testPassword"
+                //     },
+                //     "ClientCertificateMode" : "NoCertificate",
+                //     "Sni": {
```

Comment:
Offline discussion:
Throw if any of these SSL settings are enabled for an HTTP endpoint.